react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -62,6 +69,11 @@ export const Reveal3DResources = ({
     [reveal3DModels]
   );
 
+  const { data: assetMappings } = useAssetMappedNodesForRevisions(cadModelOptions);
+
+  useGenerateAssetMappingCachePerItemFromModelCache(cadModelOptions, assetMappings);
+  useGenerateNode3DCache(cadModelOptions, assetMappings);
+
```

Comment:
I don't think we should do this here. I think updating the cache should happen from within the cache itself, and not relying on these hooks getting called with the output from the cache. Sorry for not catching this before :see_no_evil:

Reply:
Hmmm if I understood your thoughts correctly, actually these two hooks are here to cache the asset mappings per item based on asset ids and node ids. The `assetMappings` as the output from the hook `const { data: assetMappings } = useAssetMappedNodesForRevisions(cadModelOptions);` above those two hooks are from the model cache which is used in different places in the app. 
The cache per item is used by rule based and any other components that doesnt need all the asset mappings per model at once and in the same order, which is a custom list of asset mappings everytime.

Reply:
After testing moving those two hooks into the main hook `useAssetMappedNodesForRevisions` to fetch the asset mappings, the Search UI start to crash and I think it is because the same main hook is called in many other components and it looks like they are trying to fetch the same items in parallel as they are not cached by the useMemo yet. So I'm thinking some alternatives but for now I believe it is safer to keep these two hooks to cache the asset mappings and nodes per item in the Reveal3DResource only or remove it entirely and them we can rethink carefully a better way to accomodate these cache hooks into the same `useAssetMappedNodesForRevisions`